src/caret_analyze/infra/lttng/records_provider_lttng.py

```diff
@@ -1241,17 +1312,56 @@ def to_records(self):
         assert self._node_path.subscription is not None and self._node_path.publisher is not None
 
         sub_records = self._provider.subscribe_records(self._node_path.subscription)
+
+        # If explicitly take message by user, there are cases that source_timestamp is 0.
+        def fill_source_timestamp_with_latest_timestamp(records):
+            source_columns = [s for s in records.columns if 'source_timestamp' in s]
+            if len(source_columns) != 1:
+                return records
+            source_column = source_columns[0]
+
+            columns = []
+            for column in records.columns:
+                columns += [ColumnValue(column)]
+
+            records_data = []
+            tmp_timestamp = 0
```

Comment:
Variable name change is recommended (`tmp_timestamp` -> `latest_timestamp`).

Reply:
Fixed in https://github.com/tier4/caret_analyze/pull/516/commits/e43e772c6edfb4cd50b46fa6d3e9a22d0b40bf52.